server.go

```diff
@@ -268,6 +276,9 @@ func (s *Server) processUnaryRPC(t transport.ServerTransport, stream *transport.
 			}
 			return
 		}
+		if EnableTracing {
```

Comment:
Check `traceInfo.tr != nil` instead. Keep only one place here consulting the `EnableTracing` global.